common/persistence/visibility/store/elasticsearch/query_interceptors.go

```diff
@@ -120,19 +148,23 @@ func (ni *nameInterceptor) Name(name string, usage query.FieldNameUsage) (string
 }
 
 func (vi *valuesInterceptor) Values(fieldName string, values ...interface{}) ([]interface{}, error) {
-	fieldType, err := vi.searchAttributesTypeMap.GetType(fieldName)
+	name := fieldName
+	if _, exists := vi.nameInterceptor.fieldTransformations[fieldName]; exists {
+		name = vi.nameInterceptor.fieldTransformations[fieldName].newField
+	}
```

Comment:
This is no-op, `name` will be the same as `fieldName` always based on this.

I don't like this. It might happen in the future of more pseudo search attribute mapping to the same field name, and this won't be able to distinguish. Can you instead change the function header to be something like `Values(name string, fieldName string, values ...any) (any, error)` instead? The caller must have the original name before mapping, so calling with this should work better.

Reply:
I cannot imagine a circumstance in which we would want to map to the same field name, but I can change this